frontend/apps/reader/modules/readerfont.lua

```diff
@@ -346,24 +336,79 @@ function ReaderFont:onAdjustFontSize(ges, direction)
     return true
 end
 
-function ReaderFont:hasFontsTestSample()
-    local font_test_sample = require("datastorage"):getSettingsDir() .. "/fonts-test-sample.html"
-    local lfs = require("libs/libkoreader-lfs")
-    return lfs.attributes(font_test_sample, "mode") == "file"
+function ReaderFont:getFontSettingsTable()
+    local settings_table = {}
+
+    if Device:isAndroid() or Device:isDesktop() or Device:isEmulator() then
+        for _, item in ipairs(require("ui/elements/font_settings"):getSystemFontMenuItems()) do
+            table.insert(settings_table, item)
+        end
+        settings_table[#settings_table].separator = true
+    end
+
+    table.insert(settings_table, {
+        text = _("Use additional fallback fonts"),
+        checked_func = function()
+            return G_reader_settings:nilOrTrue("additional_fallback_fonts")
+        end,
+        callback = function()
+        G_reader_settings:flipNilOrTrue("additional_fallback_fonts")
+            self.ui.document:setupFallbackFontFaces()
+            self.ui:handleEvent(Event:new("UpdatePos"))
+        end,
+        help_text = T(_([[
+Enable additional fallback fonts, for the most complete script and language coverage.
+These fonts will be used in this order:
+
+%1
+
+You can set a prefered fallback font with a long-press on a font name, and it will be used before these.
+If that font happens to be part of this list already, it will be used first.]]),
+            table.concat(self.ui.document.fallback_fonts, "\n")),
+        separator = true,
+    })
+
+    table.insert(settings_table, {
+        text = _("Generate font test HTML document"),
```

Comment:
I'd just leave it out, but otherwise I think HTML font test document sounds better.
```suggestion
        text = _("Generate font test document"),
```